### PR TITLE
Build, test, deploy workflows/pipelines to publish to GH Pages

### PR DESCRIPTION
**Pull request by galund**
A pipeline that appears to work well enough to deserve some more review. We won't set up DNS for the GH Pages site until we've had some content design eyes and fixed the layout.

We're deploying to https://alphagov.github.io/di-identity-vocab/

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Deploy MkDocs site to Pages
+
+on:
+  push:
+    branches: [$default-branch, deploy-to-gh-pages]
```
Comment (philandstuff) [inline .github/workflows/deploy.yml]: guessing you'll remove the second branch before merging?

Comment (galund) [reply .github/workflows/deploy.yml]: Yep, should do :-) - impossible to test without tho!

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Build
+        run: scripts/build
+
+      - name: Run all tests
+        run: scripts/test
+
+      - name: Upload HTML
+        uses: actions/upload-pages-artifact@v1.0.6
+        with:
+          path: site
```
Comment (TobySaundersGDS) [inline .github/workflows/build-test.yml:54]: Does this path match the output of the build script? I could't tell easily.

Comment (galund) [reply .github/workflows/build-test.yml:54]: Ah, yes it does, the line that does it is this https://github.com/alphagov/di-identity-vocab/blob/87ceeab90765256c54e94a8b16e9c698da5fb7f1/scripts/build#L16 and the default is "site" per https://www.mkdocs.org/user-guide/configuration/#site_dir

**Review by TobySaundersGDS — APPROVED** [review]
nice

**Review by TobySaundersGDS — APPROVED** [review]

**Pull request merged**